vm/devices/storage/scsidisk/src/lib.rs

```diff
@@ -867,6 +885,134 @@ impl SimpleScsiDisk {
         Ok(RequestParameters { tx, offset, fua })
     }
 
+    fn validate_data_cdb16_atomic(
+        &self,
+        external_data: &RequestBuffers<'_>,
+        request: &Request,
+        sector_count: u64,
+    ) -> Result<RequestParameters, ScsiError> {
+        let cdb = scsi::Cdb16Atomic::read_from_prefix(&request.cdb[..])
+            .unwrap()
+            .0; // TODO: zerocopy: use-rest-of-range (https://github.com/microsoft/openvmm/issues/759)
+        let len = cdb.transfer_blocks.get() as u64;
```

Comment:
`len` in Rust usually refers to a byte length. Consider `block_count` or something.